pkg/network/protocols/http/ebpf_ssl.go

```diff
@@ -348,11 +354,42 @@ func (o *sslProgram) Stop() {
 	o.perfHandler.Stop()
 }
 
-func addHooks(m *errtelemetry.Manager, probes []manager.ProbesSelector) func(string) error {
-	return func(libPath string) error {
-		uid := getUID(libPath)
+func addHooks(m *errtelemetry.Manager, probes []manager.ProbesSelector) func(pathIdentifier, string, string) error {
+	return func(id pathIdentifier, root string, path string) error {
+		uid := getUID(id)
+
+		elfFile, err := elf.Open(root + path)
+		if err != nil {
+			return err
+		}
+		defer elfFile.Close()
 
+		symbolsSet := make(common.StringSet, 0)
+		symbolsSetBestEffort := make(common.StringSet, 0)
 		for _, singleProbe := range probes {
+			_, isBestEffort := singleProbe.(*manager.BestEffort)
+			for _, selector := range singleProbe.GetProbesIdentificationPairList() {
+				sp := strings.Split(selector.EBPFSection, "/")
+				symbol := sp[len(sp)-1]
+				if len(symbol) == 0 {
+					continue
+				}
+				if isBestEffort {
+					symbolsSetBestEffort[symbol] = struct{}{}
+				} else {
+					symbolsSet[symbol] = struct{}{}
+				}
+			}
+		}
+		symbolMap, err := bininspect.GetAllSymbolsByName(elfFile, symbolsSet)
+		if err != nil {
+			return err
+		}
+		/* Best effort to resolve symbols, so we don't care about the error */
+		symbolMapBestEffort, _ := bininspect.GetAllSymbolsByName(elfFile, symbolsSetBestEffort)
+
+		for _, singleProbe := range probes {
+			_, isBestEffort := singleProbe.(*manager.BestEffort)
 			for _, selector := range singleProbe.GetProbesIdentificationPairList() {
```

Comment:
can we extract this to a helper function, add unit tests and document it? there is a lot of going on here and the code is a bit hard to follow :sweat:

Reply:
https://datadoghq.atlassian.net/browse/USMO-202